site/docs/glossary/types.md

```diff
@@ -13,3 +17,23 @@ SourceHash uniquely identifies the source of a deposit transaction and serves as
 ## SourceHashDomain
 
 SourceHashDomain specifies whether the deposit transaction is from a deposit on L1 or is a system transaction. You can see the type [here](https://github.com/base-org/op-viem/blob/f6595e8a0373f79e17b3c2c89f486091833e4d17/src/types/depositTransaction.ts#L5-L8).
+
+## TransactionDepositedEvent
+
+An [event](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/OptimismPortal.sol#L73C1-L73C1) emitted from a [depositTransaction](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/OptimismPortal.sol#L377) call to the [OptimismPortal](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/OptimismPortal.sol) contract, which is used to create a [DepositTransaction](#deposittransaction) on L2.
+
+## WithdrawTransaction
+
+A withdraw transaction is a special type of transaction format specific to the OP Stack. It is for creating L1 transactions from L2. The withdrawal flow consists of three parts. First, there is the withdraw initiating transaction which the user submits on L2 and emits a [MessagePassedEvent](#messagepassedevent). Second, there is the withdrawal proving transaction which is submitted on L1 by the user to prove legitimacy of the withdrawal. This transaction emits a [WithdrawalProvenEvent](#withdrawalprovenevent). Lastly, there is the withdrawal finalizing transaction which is submitted on L1 by the user after the fault challenge period has passed which emits a [WithdrawalFinalizedEvent](#withdrawalfinalizedevent) and executes the transaction on L1.
```

Comment:
```suggestion
A withdraw transaction is a special type of transaction specific to the OP Stack. It is for creating L1 transactions from L2. The withdrawal flow consists of three parts. First, there is the withdraw initiating transaction which the user submits on L2 and emits a [MessagePassedEvent](#messagepassedevent). Second, there is the withdrawal proving transaction which is submitted on L1 by the user to prove legitimacy of the withdrawal. This transaction emits a [WithdrawalProvenEvent](#withdrawalprovenevent). Lastly, there is the withdrawal finalizing transaction which is submitted on L1 by the user after the fault challenge period has passed which emits a [WithdrawalFinalizedEvent](#withdrawalfinalizedevent) and executes the transaction on L1.
```
Yea since it's l2 to l1 it's not a new tx type. I doubt we ever prioritize it at OP but ideally this would be a special tx type though IMO

Reply:
Thanks for this!